File: src/0647-回文子串/0647.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        res = 0
        for i in range(len(s)):
            res += self.extend(s, i, i)
            res += self.extend(s, i, i + 1)
        return res

    def extend(self, s, i, j):
        res = 0
        while i >= 0 and j < len(s) and s[i] == s[j]:
            i -= 1
            j += 1
            res += 1
        return res

File: src/0647-回文子串/0647.cpp
class Solution
{
public:
    int countSubstrings(string s)
    {
        int count = 0;
        for (int i = 0; i < s.length(); i++)
        {
            count += extendSubString(s, i, i);     // 奇数长度
            count += extendSubString(s, i, i + 1); // 偶数长度
        }
        return count;
    }

    int extendSubString(string s, int l, int r)
    {
        int count = 0;
        while (l >= 0 && r < s.length() && s[l] == s[r])
        {
            l--;
            r++;
            count++;
        }
        return count;
    }
};